### PR TITLE
Update example training agent and script

### PR DESCRIPTION
**Pull request by glmcdona**
Updates:
1. Split unit and city actions.
2. Update reward function to a better example that is a delta of reward and scaled reasonably. Fixes issue #83.
3. Set default example agent to inference on non-deterministic mode. Agents often get stuck when set to deterministic in inference.
2. Fix bug in unit maps where it wouldn't track nearest unit correctly.
3. Add multi-environment training command-line arg.
4. Add multi-environment evaluation metrics logging.
5. Add single-environment tensorboard game internal metrics logging.

### CODE REVIEW

File: examples/agent_policy.py [hunk 1]
```diff
@@ -437,17 +443,44 @@ def action_code_to_action(self, action_code, game, unit=None, city_tile=None, te
...
+                    unit=unit,
+                    city_id=city_tile.city_id if city_tile else None,
+                    citytile=city_tile,
+                    team=team,
+                    x=x,
+                    y=y
+                )
+
+                # If the city action is invalid, default to research action automatically
+                if not action.is_valid(game, actions_validated=[]):
```
Comment (royerk) [inline examples/agent_policy.py:460]: Should we have a `do_nothing` action for the city? There can be cases where the city should wait before spawning a unit and keep its cooldown at 0. What do you think?

**Comment by royerk** [comment]
Please let me know if you would like me (us?) to also run this update. Looks great, looking forward to have feedback on multi-envs training :heart:

File: examples/agent_policy.py (hunk 1 above)
Comment (glmcdona) [reply examples/agent_policy.py:460]: Good suggestion. A smart AI should be able to learn to use it some rare cases.

Comment (glmcdona) [reply examples/agent_policy.py:460]: Sorry, I'm just going to leave this one for future work so I don't have to retest everything. I think someone adding this to their personal scripts is a good idea, or you are welcome to add it to the example if you like!

**Comment by nosound2** [comment]
Hi @glmcdona , just a small remark from me. Many of the changes in this pull request seem to me already too deep implementation details, the kind of things everyone should decide for himself. I would have kept the repo cleaner than that. However, I am not familiar with the repository philosophy, just a first thought.

**Comment by glmcdona** [comment]
> Hi @glmcdona , just a small remark from me. Many of the changes in this pull request seem to me already too deep implementation details, the kind of things everyone should decide for himself. I would have kept the repo cleaner than that. However, I am not familiar with the repository philosophy, just a first thought.
> 
> [UPD] the new tensorboard logging stats are cool

Totally agree. Clean and simple for the example I think makes sense.

**Pull request merged**